### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by sypeer**
It is:
- Test fixes for the WIP - improving contract versions and markets creation

### CODE REVIEW

File: contracts/mock/base/EscrowMock.sol [hunk 1]
```diff
@@ -6,20 +6,29 @@ import "../../base/Escrow.sol";
...
 
-    function mockInitialize(
-        address settingsAddress,
-        address loansAddress,
-        uint256 aLoanID
-    ) external {
-        settings = SettingsInterface(settingsAddress);
-        loans = LoansInterface(loansAddress);
-        loanID = aLoanID;
+    // function mockInitialize(
```
Comment (salazarguille) [inline contracts/mock/base/EscrowMock.sol:16]: Please, remove the commented code.

File: contracts/base/Escrow/Dapps/Uniswap.sol [hunk 2]
```diff
@@ -108,8 +108,8 @@ contract Uniswap is IUniswap, BaseEscrowDapp {
         require(amounts.length == path.length , "UNISWAP_ERROR_SWAPPING");
         uint256 amountReceived = amounts[amounts.length - 1];
 
-        _tokenUpdated(source);
-        _tokenUpdated(destination);
+        // _tokenUpdated(source);
```
Comment (salazarguille) [inline contracts/base/Escrow/Dapps/Uniswap.sol:111]: Please, add a TODO

File: test/atm/ATMFactoryCreateATMTest.js [hunk 3]
```diff
@@ -34,27 +32,23 @@ contract("ATMFactoryCreateATMTest", function(accounts) {
...
                     name,
                     symbol,
                     decimals,
                     cap,
                     maxVesting,
-                    atmGovernance.address,
-                    atmToken.address,
                     {from : sender }
                 );
+                console.log("RESULT>>>", result);
```
Comment (salazarguille) [inline test/atm/ATMFactoryCreateATMTest.js:45]: Please, remove console.log

File: test/base/BaseLoansCreateEscrowTest.js [hunk 4]
```diff
@@ -59,19 +82,21 @@ contract('BaseLoansCreateEscrowTest', function (accounts) {
...
         // If the test must fail from the resulting transaction, then the call to the function (not a transaction) will also fail but as a different Error object
-        if (!mustFail)
+        if (!mustFail) {
           escrowAddress = await loans.externalCreateEscrow.call(loanID);
+        }
+
         const result = await loans.externalCreateEscrow(loanID);
         // TODO fix unit test.
-
+        console.log("RESULT>>>>", result);
```
Comment (salazarguille) [inline test/base/BaseLoansCreateEscrowTest.js:91]: Please, remove console.log.

File: test/base/BaseLoansCreateEscrowTest.js [hunk 5]
```diff
@@ -59,19 +82,21 @@ contract('BaseLoansCreateEscrowTest', function (accounts) {
...
         // Assertions
         assert(!mustFail);
 
         await escrowFactory
           .escrowCreated(result, EscrowFactory)
           .emitted(borrower, loans.address, loanID.toString(), escrowAddress);
       } catch (error) {
-        console.log(error);
+        console.log("ERROR>>>>", error);
```
Comment (salazarguille) [inline test/base/BaseLoansCreateEscrowTest.js:99]: Please, remove console.log.

File: test/base/EscrowCallDappTest.js [hunk 6]
```diff
@@ -25,9 +25,9 @@ contract('EscrowCallDappTest', function (accounts) {
     loans = await Mock.new();
     escrow = await Escrow.new();
     instance = await EscrowFactory.new();
-    await instance.initialize(settingsInstance.address, escrow.address);
+    await instance.initialize(settingsInstance.address);
 
-    await settingsInstance.setEscrowFactory(instance.address)
+    // await settingsInstance.setEscrowFactory(instance.address)
```
Comment (salazarguille) [inline test/base/EscrowCallDappTest.js:30]: Please, remove the commented code.

File: test/base/EscrowCallDappTest.js [hunk 7]
```diff
@@ -66,12 +68,13 @@ contract('EscrowCallDappTest', function (accounts) {
...
         // Invocation
-        const result = await escrow.callDapp(dappData);
-
+        const result = await escrow.callDapp(dappData, { from: owner });
+        
         // Assertions
         assert(!mustFail);
         assert(result);
       } catch (error) {
+        // console.log("ERROR>>>>", error);
```
Comment (salazarguille) [inline test/base/EscrowCallDappTest.js:77]: Please, remove the commented code.

File: test/base/EscrowFactoryCreateEscrowTest.js [hunk 8]
```diff
@@ -17,16 +17,14 @@ contract('EscrowFactoryCreateEscrowTest', function (accounts) {
...
   beforeEach(async () => {
     settingsInstance = await createTestSettingsInstance(Settings, { from: owner, Mock });
     loans = await Loans.new();
-    escrowLibrary = await Escrow.new();
     instance = await EscrowFactory.new();
 
-    await instance.initialize(settingsInstance.address, escrowLibrary.address);
-    await settingsInstance.setEscrowFactory(instance.address);
+    await instance.initialize(settingsInstance.address);
+    // await settingsInstance.setEscrowFactory(instance.address);
```
Comment (salazarguille) [inline test/base/EscrowFactoryCreateEscrowTest.js:27]: Please, remove the commented code.

File: test/base/EscrowFactoryCreateEscrowTest.js [hunk 9]
```diff
@@ -59,6 +57,7 @@ contract('EscrowFactoryCreateEscrowTest', function (accounts) {
         // Assertions
         assert(!mustFail);
       } catch (error) {
+        console.log("ERROR>>>>", error);
```
Comment (salazarguille) [inline test/base/EscrowFactoryCreateEscrowTest.js:60]: Please, remove the console.log.

File: test/base/EscrowIsBorrowerTest.js [hunk 10]
```diff
@@ -60,18 +59,20 @@ contract('EscrowIsBorrowerTest', function (accounts) {
...
         assert(result);
+        assert.equal(
+          result,
+          sender,
+          "Escrow is not borrower"
+        )
       } catch (error) {
         assert(mustFail, error);
         assert(error);
-        assert.equal(error.reason, expectedErrorMessage);
```
Comment (salazarguille) [inline test/base/EscrowIsBorrowerTest.js:74]: Why we are not validating the error message?

File: test/base/EtherCollateralLoansDepositCollateralTest.js [hunk 11]
```diff
@@ -27,15 +26,13 @@ contract('EtherCollateralLoansDepositCollateralTest', function (accounts) {
...
-        marketsInstance = await Mock.new();
         atmSettingsInstance = await Mock.new();
         instance = await Loans.new();
         await instance.initialize(
             oracleInstance.address,
             lendingPoolInstance.address,
             loanTermsConsInstance.address,
             settingsInstance.address,
-            marketsInstance.address,
             atmSettingsInstance.address,
```
Comment (salazarguille) [inline test/base/EtherCollateralLoansDepositCollateralTest.js:36]: I think we removed the atmSettings from Loans.initialize(...) due to it is in the settings contract.

File: test/base/EtherCollateralLoansRepayTest.js [hunk 12]
```diff
@@ -45,7 +44,6 @@ contract('EtherCollateralLoansRepayTest', function (accounts) {
             lendingPoolInstance.address,
             loanTermsConsInstance.address,
             settingsInstance.address,
-            collateralTokenInstance.address,
             atmSettingsInstance.address,
```
Comment (salazarguille) [inline test/base/EtherCollateralLoansRepayTest.js:47]: I  think  the last param in the Loans.initialize is the collateral token.

File: test/base/EtherCollateralLoansWithdrawCollateralTest.js [hunk 13]
```diff
@@ -36,15 +35,13 @@ contract('EtherCollateralLoansWithdrawCollateralTest', function (accounts) {
...
-        marketsInstance = await Mock.new();
         atmSettingsInstance = await Mock.new();
         instance = await Loans.new();
         await instance.initialize(
             oracleInstance.address,
             lendingPoolInstance.address,
             loanTermsConsInstance.address,
             settingsInstance.address,
-            marketsInstance.address,
             atmSettingsInstance.address,
```
Comment (salazarguille) [inline test/base/EtherCollateralLoansWithdrawCollateralTest.js:45]: The same, I think atmSettingsInstance should be collateralTokenInstance.

File: test/base/LoanTermsConsensusProcessRequestTest.js [hunk 14]
```diff
@@ -138,12 +138,13 @@ contract('LoanTermsConsensusProcessRequestTest', function (accounts) {
...
             await instance.initialize(loansContract, settings.address, markets.address);
+            const txConfig = { from: loansContract };
 
-            await instance.addSigner(nodeOne)
-            await instance.addSigner(nodeTwo)
-            await instance.addSigner(nodeThree)
-            await instance.addSigner(nodeFour)
-            await instance.addSigner(nodeSix)
+            await instance.addSigner(nodeOne, txConfig);
+            await instance.addSigner(nodeTwo, txConfig);
```
Comment (salazarguille) [inline test/base/LoanTermsConsensusProcessRequestTest.js:144]: why do we need to add the txConfig?

File: contracts/mock/base/EscrowMock.sol (hunk 1 above)
Comment (sypeer) [reply contracts/mock/base/EscrowMock.sol:16]: Removed

File: contracts/base/Escrow/Dapps/Uniswap.sol (hunk 2 above)
Comment (sypeer) [reply contracts/base/Escrow/Dapps/Uniswap.sol:111]: Added

File: test/atm/ATMFactoryCreateATMTest.js (hunk 3 above)
Comment (sypeer) [reply test/atm/ATMFactoryCreateATMTest.js:45]: Removed

File: test/base/BaseLoansCreateEscrowTest.js (hunk 4 above)
Comment (sypeer) [reply test/base/BaseLoansCreateEscrowTest.js:91]: Removed

File: test/base/BaseLoansCreateEscrowTest.js (hunk 5 above)
Comment (sypeer) [reply test/base/BaseLoansCreateEscrowTest.js:99]: Removed

File: test/base/EscrowCallDappTest.js (hunk 6 above)
Comment (sypeer) [reply test/base/EscrowCallDappTest.js:30]: Removed

File: test/base/EscrowCallDappTest.js (hunk 7 above)
Comment (sypeer) [reply test/base/EscrowCallDappTest.js:77]: Removed

File: test/base/EscrowFactoryCreateEscrowTest.js (hunk 8 above)
Comment (sypeer) [reply test/base/EscrowFactoryCreateEscrowTest.js:27]: Removed

File: test/base/EscrowFactoryCreateEscrowTest.js (hunk 9 above)
Comment (sypeer) [reply test/base/EscrowFactoryCreateEscrowTest.js:60]: Removed

File: test/base/EtherCollateralLoansDepositCollateralTest.js (hunk 11 above)
Comment (sypeer) [reply test/base/EtherCollateralLoansDepositCollateralTest.js:36]: EtherCollateralLoans still has 5 params (1 that is empty). Needs to be changed then

File: test/base/EtherCollateralLoansRepayTest.js (hunk 12 above)
Comment (sypeer) [reply test/base/EtherCollateralLoansRepayTest.js:47]: Changed

File: test/base/EtherCollateralLoansWithdrawCollateralTest.js (hunk 13 above)
Comment (sypeer) [reply test/base/EtherCollateralLoansWithdrawCollateralTest.js:45]: Changed

File: test/base/LoanTermsConsensusProcessRequestTest.js (hunk 14 above)
Comment (sypeer) [reply test/base/LoanTermsConsensusProcessRequestTest.js:144]: Without the txConfig all the tests were failing with a VM Exception error - reason given - Ownable: caller is not the owner

Currently, with the txConfig, there are 2 passing and 8 failing mostly with a RESPONSE_EXPIRED error

**Pull request merged**